source/patterns/@aws-solutions-constructs/core/test/test-helper.ts

```diff
@@ -178,3 +177,26 @@ export function expectNonexistence(stack: Stack, type: string, props: object) {
   const shouldFindNothing = Template.fromStack(stack).findResources(type, props);
   expect(Object.keys(shouldFindNothing).length).toEqual(0);
 }
+
+// private helper class to suppress the standard cfn nag warnings for lambda functions used in integ tests
+class CfnNagLambdaAspect implements IAspect {
```

Comment:
This is a very different approach than suppressAutoDeleteHandlerWarnings (that I'm still trying to understand). As "the same is better than better", my inclination is to use similar approaches.

If we go with this new approach, do. we update suppressAutoDeleteHandlerWarnings to work the same way?

Reply:
Yeah so my initial approach was to do it the same way we always have. Unfortunately there is an [open issue](https://github.com/aws/aws-cdk/issues/18357) as to why its proving difficult to target the specific lambda function that is created behind the scenes when enabling the autoDeleteObjects flag on the bucket construct. This led me to go with the aspect approach, as it will find any lambda function on the entire stack and apply the suppressions.